src/EVM/Expr.hs

```diff
@@ -1350,35 +1350,20 @@ constFoldProp ps = oneRun ps (ConstState mempty True)
         PBool False -> put $ ConstState {canBeSat=False, values=mempty}
         _ -> pure ()
 
--- Concretize Keccak Expressions until fixed-point.
--- Can either simplify before, or not
-concKeccakSimpExpr :: Bool -> Expr a -> Expr a
-concKeccakSimpExpr simp orig = final
-  where
-    simplified = if simp then simplify orig else orig
-    conc = mapExpr concKeccakSimpOnePass simplified
-    final = if conc == orig then conc
-                            else concKeccakSimpExpr simp conc
-
-concKeccakSimps :: Bool -> [Prop] -> [Prop]
-concKeccakSimps simp orig = final
-  where
-    conc = simplifyProps $ map (concKeccakSimp simp) orig
-    final = if conc == orig then conc
-                            else concKeccakSimps simp conc
-
-concKeccakSimp :: Bool -> Prop -> Prop
-concKeccakSimp simp orig = final
-  where
-    simplified = if simp then simplifyProp orig else orig
-    conc = mapProp concKeccakSimpOnePass simplified
-    final = if conc == orig then conc
-                            else concKeccakSimp simp conc
-
-concKeccakSimpOnePass :: Expr a -> Expr a
-concKeccakSimpOnePass (Keccak (ConcreteBuf bs)) = Lit (keccak' bs)
-concKeccakSimpOnePass orig@(Keccak (CopySlice (Lit 0) (Lit 0) (Lit 64) orig2@(WriteWord (Lit 0) _ (ConcreteBuf bs)) (ConcreteBuf ""))) =
+-- Concretize & simplify Keccak expressions until fixed-point.
+concKeccakSimpExpr :: Expr a -> Expr a
+concKeccakSimpExpr orig = untilFixpoint ((mapExpr concKeccakOnePass) . simplify) orig
+
+concKeccakProps :: [Prop] -> [Prop]
```

Comment:
```suggestion
concKeccakSimpProps :: [Prop] -> [Prop]
```

just for consistency...

Reply:
Oh wow, this was a bug actually. There shouldn't have been a `simplifyProps $` in front. So the name was correct, and there ought to be no `Simp` there. I now removed the `simplifyProps $` and the name is now correct :)